src/main/java/uk/gov/hmcts/reform/cpo/validators/annotation/ValidCpoId.java

```diff
@@ -0,0 +1,25 @@
+package uk.gov.hmcts.reform.cpo.validators.annotation;
+
+import uk.gov.hmcts.reform.cpo.validators.IdsValidator;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+@Constraint(validatedBy = IdsValidator.class)
+@Target({PARAMETER, FIELD})
+@Retention(RUNTIME)
+public @interface ValidCpoId {
+
+    String message() default "Error messages in id validator, the constraint is violated";
```

Comment:
is this the error the endpoint client would get in case the case id is not valid? If yes, can we provide a better message please?

Reply:
DONE I added a default message.